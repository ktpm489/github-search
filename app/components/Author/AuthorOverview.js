import React from 'react';
import PropTypes from 'prop-types';
import { View, StyleSheet } from 'react-native';
import { Avatar } from 'react-native-elements';
import LabeledIcon from '../Primitives/LabeledIcon';

const AuthorOverview = ({
  company,
  location,
  followerCount,
  websiteUrl,
  repoCount,
  organizations,
}) => {
  const data = [
    { iconType: 'material', iconName: 'group', item: company },
    { iconType: 'entypo', iconName: 'link', item: websiteUrl },
    { iconType: 'evil', iconName: 'location', item: location },
    {
      iconType: 'entypo',
      iconName: 'github',
      item: `${followerCount} followers`,
    },
    {
      iconType: 'entypo',
      iconName: 'log-out',
      item: `${repoCount} repositories`,
    },
  ];

  return (
    <View style={styles.container}>
      {data.map(
        (item, index) =>
          item.item && (
            <LabeledIcon
              {...item}
              iconSize={30}
              iconColor={'#222'}
              key={index}
            />
          )
      )}
      <View style={{ flex: 1, flexDirection: 'row' }}>
        {organizations.map(({ node: { avatarUrl, name } }) => (
          <Avatar
            key={name}
            height={50}
            width={50}
            source={{ uri: avatarUrl }}
            containerStyle={{
              margin: 10,
            }}
          />
        ))}
      </View>
    </View>
  );
};

AuthorOverview.propTypes = {
  company: PropTypes.string,
  location: PropTypes.string,
  followerCount: PropTypes.number,
  websiteUrl: PropTypes.string,
  repoCount: PropTypes.number,
  organizations: PropTypes.arrayOf(
    PropTypes.shape({
      node: PropTypes.shape({
        avatarUrl: PropTypes.string,
        name: PropTypes.string.isRequired,
      }),
    })
  ),
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});

export default AuthorOverview;
