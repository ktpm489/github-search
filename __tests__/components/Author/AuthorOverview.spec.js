import renderer from 'react-test-renderer';
import React from 'react';
import AuthorOverview from '../../../app/components/Author/AuthorOverview';

describe('<AuthorOverview />', () => {
  it('renders correctly with all required props', () => {
    const tree = renderer
      .create(
        <AuthorOverview
          company="Formidable"
          location="Seattle, WA"
          followerCount={37}
          websiteUrl="https://parkerziegler.com"
          repoCount={22}
          organizations={[{ node: { name: 'Formidable' } }]}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
