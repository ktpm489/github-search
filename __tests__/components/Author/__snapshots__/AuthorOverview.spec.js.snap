// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthorOverview /> renders correctly with all required props 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "margin": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#222",
          "fontSize": 15,
          "marginLeft": 10,
        }
      }
    >
      Formidable
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "margin": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#222",
          "fontSize": 15,
          "marginLeft": 10,
        }
      }
    >
      https://parkerziegler.com
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "margin": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#222",
          "fontSize": 15,
          "marginLeft": 10,
        }
      }
    >
      Seattle, WA
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "margin": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#222",
          "fontSize": 15,
          "marginLeft": 10,
        }
      }
    >
      37 followers
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "margin": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#222",
          "fontSize": 15,
          "marginLeft": 10,
        }
      }
    >
      22 repositories
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      height={50}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 50,
            "paddingBottom": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "width": 50,
          },
          undefined,
          Object {
            "margin": 10,
          },
        ]
      }
      width={50}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "rgba(0,0,0,0.2)",
              "bottom": 0,
              "flex": 1,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Array [
              Object {
                "height": 50,
                "width": 50,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
